static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -93,25 +98,23 @@ const MemoizedWidgetCardChartContainer = React.memo(
   (prevProps, props) => {
     return (
       props.selection === prevProps.selection &&
-      props.location.query[WidgetViewerQueryField.QUERY] ===
-        prevProps.location.query[WidgetViewerQueryField.QUERY] &&
-      props.location.query[WidgetViewerQueryField.SORT] ===
-        prevProps.location.query[WidgetViewerQueryField.SORT] &&
-      props.location.query[WidgetViewerQueryField.WIDTH] ===
-        prevProps.location.query[WidgetViewerQueryField.WIDTH]
+      (props.location.query[WidgetViewerQueryField.SORT] ===
+        prevProps.location.query[WidgetViewerQueryField.SORT] ||
+        (props.widget.displayType !== DisplayType.TOP_N &&
+          props.widget.limit !== undefined)) &&
+      isEqual(props.chartZoomOptions, prevProps.chartZoomOptions)
```

Comment:
These queries are rough to read, might be nice to have some interim variables here

Reply:
👍  cleaned this up a bit